fem/coefficient.hpp

```diff
@@ -942,6 +1148,39 @@ class OuterProductCoefficient : public MatrixCoefficient
 public:
    OuterProductCoefficient(VectorCoefficient &A, VectorCoefficient &B);
 
+   void SetACoef(VectorCoefficient &A) { a = &A; }
+   VectorCoefficient * GetACoef() const { return a; }
+
+   void SetBCoef(VectorCoefficient &B) { b = &B; }
+   VectorCoefficient * GetBCoef() const { return b; }
+
+   virtual void Eval(DenseMatrix &M, ElementTransformation &T,
+                     const IntegrationPoint &ip);
+};
+
+/// Matrix coefficient defined as -a k x k x, for a vector k and scalar a
+class CrossCrossCoefficient : public MatrixCoefficient
```

Comment:
Should we check if the vector dimension is 3? Or does the class also work for other dimensions?

Reply:
Good point!  I'll clarify.